rlog/2024-06-24-gsub-largemsg-improvements.mdx

```diff
@@ -0,0 +1,396 @@
+typically result in longer than expected message dissemination times for large messages (due to cold connections). 
+It is also worth mentioning that some TCP variants reset their $C_{wnd}$ after different periods of inactivity.
+
+3. Theoretically, the message transmission time to D peers $(\tau_D)$ remains the same, 
+even if the message is relayed sequentially to all peers or a simultaneous transmission is carried out.
+However, sequential transmissions finish early for individual peers, allowing them to relay early.
+It may result in quicker network-wide message dissemination. 
+
+4. A realistic network comprises nodes with dissimilar capabilities (bandwidth, link latency, compute, etc.).
+As the message disseminates, it's not uncommon for some peers to receive it much earlier than others. 
+Early gossip (IHAVE announcements) may bring in many IWANT requests to the early receivers (even from peers already receiving the same message), 
+which adds to their workload.  
+
+5. A busy peer (with a sizeable outgoing message queue) will enqueue (or simultaneously transfer) newly scheduled outgoing messages. 
+As a result, already scheduled messages are prioritized over peers' locally published messages, 
+introducing a significant initial delay to the locally published messages. 
+Enqueuing IWANT replies to the outgoing message queue can further exaggerate the problem. 
+The lack of adaptiveness and standardization in outgoing message prioritization are key factors that can lead to noticeable inconsistency 
+in message dissemination latency at each hop, even in similar network conditions.
+
+6. Message size directly contributes to peers' workloads in terms of processing and transmission time.
+It also raises the probability of simultaneous redundant transmissions to the same peer, 
+resulting in bandwidth wastage, congestion, and slow message propagation to the network. 
+Moreover, the benefits of sequential message relaying can be compromised by prioritizing slow (or busy) peer(s).
+
+7. Most use cases necessitate validating received messages before forwarding them to the next-hop peers.
+For a higher message transfer time $(\tau )$, this store-and-forward delay accumulates across the hops traveled by the message.  
+
+## Possible Improvements
+
+### 1. Minimizing Transfer Time for Large Messages
+The impact of message size and achievable data rate on message transmit time $\tau$ is crucial, 
+as this time accumulates due to the store-and-forward delay introduced at intermediate hops. 
+Some possible improvements minimizing overall message dissemination latency include:
+
+#### a. Message Fragmentation
+In a homogeneous network, network-wide message dissemination time (ignoring any processing delays) 
+can be simplified to roughly $\delta \approx \delta_{Tx} + P_h$, where $\delta_{Tx}$ represents accumulative message transmit time denoted as $\delta_{Tx} = \frac{S}{R} \times h$, 

```

Comment:
Please define P_h and R in this context.

Reply:
bd3f3d9c0f32350e07e987ff0ffc46ed9e3ba174